pkg/flist/cleanup.go

```diff
@@ -139,9 +139,31 @@ func (f *flistModule) cleanUnusedMounts() error {
 		}
 
 		if err := os.Remove(path); err != nil {
-			log.Error().Err(err).Msgf("failed to clean mountpoint %s", path)
+			log.Debug().Msgf("failed to remove path, trying to forcibly clean up : %+v", path)
+			if err := f.forceUnmountAndRemove(path); err != nil {
+				log.Error().Err(err).Msgf("failed to clean mountpoint %s", path)
+			}
 		}
 	}
 
 	return nil
 }
+
+func (f *flistModule) forceUnmountAndRemove(path string) error {
+	// Try normal unmount first
+	err := f.system.Unmount(path, 0)
+	if err != nil {
+		log.Warn().Err(err).Msgf("normal unmount failed for %s, trying lazy unmount", path)
+
+		// Try lazy unmount (MNT_DETACH)
+		err = syscall.Unmount(path, syscall.MNT_DETACH)
```

Comment:
the above `f.system.unmount` already do `syscall.unmount` internally not sure why we need it here?